test_discovery.py

```diff
@@ -159,22 +159,27 @@ def test_all(test_file):
 
         y_true += y_true_new
         y_pred += y_pred_new
-        
+
         # include test filename, test number, test name, transcript, and time in ms
-        timing_list.append(TimingResult(test_file, test_no, test_dict["test"], test_dict["transcript"], time_elapsed))
+        timing_list.append(
+            TimingResult(
+                test_file, test_no, test_dict["test"], test_dict["transcript"],
+                time_elapsed
+            )
+        )
 
         failed_tests += entity_results['failure']
         failed_tests += intent_results['failure']
-        
+
         total_errors += entity_results['total_errors']
         total_char_errors += entity_results['total_char_errors']
         total_characters += entity_results['total_characters']
 
         test_failures.append(entity_results['test_failures'])
         test_failures.append([intent_results.get('test_failures', None)])
-            
+
         output_dict[test_no] = intent_results
-    
+
     test_failures = [i for s in test_failures for i in s]
```

Comment:
```suggestion
    test_failures = [i for s in test_failures for i in s if i]
```
if you added `if i`, could you avoid the second for loop? 
name variables :)

Reply:
The point of this is to flatten the nested list. Not sure the named variables add anything here.

Reply:
No, the loop can’t be avoided by putting an if statement. This scheme is simply to flatten the nested lists.